src/jit/emitxarch.cpp

```diff
@@ -1405,10 +1212,7 @@ inline bool hasCodeMR(instruction ins)
 
 inline size_t insCodeMR(instruction ins)
 {
-    assert((unsigned)ins < _countof(insCodesMR));
-    assert((insCodesMR[ins] != BAD_CODE));
-
-    return insCodesMR[ins];
+    return InstructionInfo::getEncoding(ins, InsEncodingFmt_mr);
```

Comment:
I'm worried that with all of these functions, we're going from a simple array access, to a double array access: one to find the instruction info array element, and one to access the encoding.

It's my assumption that these functions are hot, since I believe the encoder is hot, so giving up this performance seems bad.

Reply:
Its accessing the field of a struct in a constant array. The compiler should be optimizing this to be a single lookup.

Reply:
Also, to explicitly call it out, the `encoding` field is a C-style inline array, so there is no second indirection there.

Reply:
What really happens is that we'll end up with 1 or 2 extra instructions per array access to handle the `sizeof(InstructionInfo)` multiplication required for indexing. These instructions did not exist before as those parallel arrays were primitive arrays and indexing was handled entirely by addressing mode.